### PR TITLE
Replace multistage with vendored go

### PR DESCRIPTION
**Pull request by qinqon**
**What this PR does / why we need it**:
We are already not using multistage to run suff like vet or goimports, this PR remove it alltogether and uses a installed golang version that uses go.mod version to select it, it reduces build time and indirections.                                         
                                                                                
**Special notes for your reviewer**:                                            
                                                                                
**Release note**:                                                               
                                                                             
                                                                                
```release-note                                                                 
Golang tools vendored and multistage build removed.                                                                          
```

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# We cannot call `docker build` out of top dir since this is controlled by
```
Comment (phoracek) [inline .dockerignore:1]: Wasn't this added in a different PR?

File: Makefile [hunk 2]
```diff
@@ -34,18 +34,26 @@ E2E_SUITES = \
 	test/e2e/lifecycle \
 	test/e2e/workflow
 
-OPERATOR_SDK ?= build/_output/bin/operator-sdk
+BIN_DIR = $(CURDIR)/build/_output/bin/
```
Comment (phoracek) [inline Makefile:37]: Please add a commit message explaining why

**Comment by phoracek** [comment]
/override pull-e2e-cluster-network-addons-operator-workflow-k8s

Not related

File: .dockerignore (hunk 1 above)
Comment (qinqon) [reply .dockerignore:1]: Not at this project, but I can remove it from here.

Comment (qinqon) [reply .dockerignore:1]: Not at this project, but I can remove it from here.

**Pull request merged**